x/ibc/07-tendermint/update.go

```diff
@@ -0,0 +1,69 @@
+package tendermint
+
+import (
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	commitment "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+)
+
+// CheckValidityAndUpdateState checks if the provided header is valid and updates
+// the consensus state if appropriate. It returns an error if:
+// - the client or header provided are not parseable to tendermint types
+// - the header is invalid
+// - header height is lower than the latest client height
+// - header valset commit verification fails
+func CheckValidityAndUpdateState(
+	clientState clientexported.ClientState, header clientexported.Header, chainID string,
+) (clientexported.ClientState, clientexported.ConsensusState, error) {
+	tmClientState, ok := clientState.(ClientState)
+	if !ok {
+		return nil, nil, sdkerrors.Wrap(
+			clienttypes.ErrInvalidClientType, "light client is not from Tendermint",
+		)
+	}
+
+	tmHeader, ok := header.(Header)
+	if !ok {
+		return nil, nil, sdkerrors.Wrap(
+			clienttypes.ErrInvalidHeader, "header is not from Tendermint",
+		)
+	}
+
+	if err := checkValidity(tmClientState, tmHeader, chainID); err != nil {
+		return nil, nil, err
+	}
+
+	tmClientState, consensusState := update(tmClientState, tmHeader)
```

Comment:
In the spec, this is set to the store, not returned, same question as above.

Reply:
blocked by #5502

Reply:
Who is going to do this work and should we create an issue for this?

Reply:
> Who is going to do this work and should we create an issue for this?

Idk, already have too much stuff on my plate lol. Let's move this convo to the issue above